cmd/eval_test.go

```diff
@@ -676,6 +676,38 @@ func removeBuiltin(builtins []*ast.Builtin, name string) []*ast.Builtin {
 	return cpy
 }
 
+// Nearly identical to TestEvalWithOptimizeBundleData, but uses
+// Rego entrypoint annotations instead of explicitly providing
+// the entrypoints as CLI arguments.
+func TestEvalWithRegoEntrypointAnnotations(t *testing.T) {
+	files := map[string]string{
+		"test.rego": `
+package test
+default p = false
+# METADATA
+# entrypoint: true
+p { q }
+q { input.x = data.foo }`,
+		"data.json": `
+			{"foo": 1}`,
```

Comment:
[nit] Funky indentation for a verbatim string. But it doesn't matter, of course.

Reply:
During development, I had issues around metadata comments not being allowed to have spaces in front of them.
Example from `cmd/build_test` after putting "proper" indentation around the verbatim string:
```
Running tool: /usr/local/go/bin/go test -timeout 30s -tags opa_wasm,slow -run ^TestBuildPlanWithRegoEntrypointAnnotations$ github.com/open-policy-agent/opa/cmd

--- FAIL: TestBuildPlanWithRegoEntrypointAnnotations (0.00s)
    --- FAIL: TestBuildPlanWithRegoEntrypointAnnotations/annotated_entrypoint (0.00s)
        /dev-folder/opa/cmd/build_test.go:348: load error: 1 error occurred during loading: /tmp/opa_test3450830366/test.rego:2: rego_parse_error: expected METADATA block, found whitespace
            	# METADATA
            	^
FAIL
FAIL	github.com/open-policy-agent/opa/cmd	0.012s
FAIL
```

Reply:
The error is caused by [this check](https://github.com/open-policy-agent/opa/blob/482769dd12a1f346780ff0a26ccf2c50480f9547/ast/parser.go#L2146-L2148) in `parser.go`. I'm not sure why we can't have whitespace before a metadata annotation comment, but it seems unnecessarily limiting.  :shrug:

Reply:
I've just been referring to the data.json. but you've got a point there, I suppose.

Reply:
Ah, okay. The `data.json` indentation is easy enough to fix, and I'll roll that into my next batch of changes. Addressing the "whitespace breakage" issue is definitely material for a different PR if it's not intentional behavior.